### PR TITLE
[tests-only]unshare without permissions to share

### PR DESCRIPTION
**Pull request by swoichha**
### Description
This PR adds tests for deleting a share after changing permission of the shared file.

### Related Issue
https://github.com/owncloud/client/issues/7423

### CODE REVIEW

File: test/gui/tst_sharing/test.feature [hunk 1]
```diff
@@ -53,3 +53,36 @@ Feature: Sharing
         And the user toggles the password protection using the client-UI
         And the user toggles the password protection using the client-UI
         Then the progress indicator should not be visible in the client-UI
+
+    Scenario: unshare a reshared file
+        Given the setting "shareapi_auto_accept_share" on the server of app "core" has been set to "no"
+    	And the administrator on the server has set the default folder for received shares to "Shares"
```
Comment (dpakach) [inline test/gui/tst_sharing/test.feature:59]: fix indentation here

File: test/gui/shared/steps/steps.py [hunk 2]
```diff
@@ -350,3 +350,16 @@ def step(context):
...
     )
+  
+@When('user "|any|" opens the sharing dialog of "|any|" to delete the folder shared using the client-UI')
+def step(context, receiver, resource):
+    resource = substituteInLineCodes(context, resource).replace('//','/')
+    waitFor(lambda: isFolderSynced(resource), context.userData['clientSyncTimeout'] * 1000)
+    waitFor(lambda: shareResource(resource), context.userData['clientSyncTimeout'] * 1000)   
+    
+
+@Then('"|any|" should be displayed')
```
Comment (dpakach) [inline test/gui/shared/steps/steps.py:361]: this step is very generic and I suspect cannot be used everywhere. Can you make it a bit more descriptive?

File: test/gui/tst_sharing/test.feature (hunk 1 above)
Comment (swoichha) [reply test/gui/tst_sharing/test.feature:59]: This is weird.  Locally the indention is fine but git is showing the wrong indentation.

**Comment by SwikritiT** [comment]
LGTM :+1:

**Review by dpakach — COMMENTED** [review]
lgtm :+1:

File: test/gui/tst_sharing/test.feature (hunk 1 above)
Comment (individual-it) [reply test/gui/tst_sharing/test.feature:59]: check tabs vs spaces

File: test/gui/shared/steps/steps.py [hunk 3]
```diff
@@ -350,3 +350,16 @@ def step(context):
         waitForObjectExists(names.oCC_ShareLinkWidget_checkBox_password_QProgressIndicator).visible,
         False
     )
+  
+@When('user "|any|" opens the sharing dialog of "|any|" to delete the folder shared using the client-UI')
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:354]: it really does not matter in this step for what reason the user opens the sharing dialog. the reason and what will happen in that dialog will be determent in the next steps

```suggestion
@When('user "|any|" opens the sharing dialog of "|any|" using the client-UI')
```

**Review by individual-it — CHANGES_REQUESTED** [review]

**Review by individual-it — APPROVED** [review]

**Pull request merged**